src/main/kotlin/no/nav/tilleggsstonader/sak/utbetaling/iverksetting/IverksettStatusService.kt

```diff
@@ -23,13 +23,29 @@ class IverksettStatusService(
     @Transactional
     fun hentStatusOgOppdaterAndeler(eksternFagsakId: Long, behandlingId: UUID, iverksettingId: UUID) {
         val status = iverksettClient.hentStatus(eksternFagsakId, behandlingId, iverksettingId)
-        if (status != IverksettStatus.OK) {
-            throw TaskExceptionUtenStackTrace("Status fra oppdrag er ikke ok, status=$status")
+        val statusIverksetting = when (status) {
+            IverksettStatus.OK -> StatusIverksetting.OK
+            IverksettStatus.OK_UTEN_UTBETALING -> StatusIverksetting.OK_UTEN_UTBETALING
+            else -> throw TaskExceptionUtenStackTrace("Status fra oppdrag er ikke ok, status=$status")
         }
 
         val andeler = hentAndelerForIverksettingId(behandlingId, iverksettingId)
         logger.info("Oppdaterer ${andeler.size} andeler med iverksettingId=$iverksettingId")
-        andelTilkjentYtelseRepository.updateAll(andeler.map { it.copy(statusIverksetting = StatusIverksetting.OK) })
+        validerStatus(status, andeler)
+
+        andelTilkjentYtelseRepository.updateAll(andeler.map { it.copy(statusIverksetting = statusIverksetting) })
+    }
+
+    private fun validerStatus(
+        status: IverksettStatus,
+        andeler: List<AndelTilkjentYtelse>,
+    ) {
+        feilHvis(status == IverksettStatus.OK_UTEN_UTBETALING && (andeler.size != 1 || andeler.single().beløp != 0)) {
```

Comment:
Når forventer vi at andeler inneholder 1 andel?

Reply:
Når man kun iverksatt en 0-andel `hentAndelerForIverksettingId` henter ut andelene for iverksettingen som skal sjekkes status på

Reply:
Åja, tenkte dette var noen andeler som dp-iverksett returnerte. Men ser at vi henter dem fra egen db.